python/mach_bootstrap.py

```diff
@@ -216,7 +216,7 @@ def bootstrap(topdir):
 
     # We don't support paths with spaces for now
     # https://github.com/servo/servo/issues/9442
```

Comment:
Unfortunately I don't think we can close that bug just yet. I tested on NixOS and jemallocator's build script fails because the configure script errors with 'configure: error: Prefix should not contain spaces'.

Reply:
Hrm. In this case the bug is already closed, so maybe we should open a new bug (or a bug in the jemallocator repository) and reference it here.

Reply:
Since this is about the PREFIX too, I bet we could move this check to the NixOS only part of the script too.

Reply:
It looks like #9442 is just the window's specific bug while #9616 is for all platforms. 

> Since this is about the PREFIX, I bet we could move this check to the NixOS only part of the script too.

I haven't investigated the issue but the failure happens when build.rs of jemallocator runs configure for it's native library compilation. My understanding is that  `prefix` is a generic parameter in configure scripts used to point to the build directory, which in jemallocator's case is inside `target/release../`  and thus `prefix` have a space if any component of the path to Servo repo contains a space. It doesn't look specific to NixOS and seems like it would impact all Linux distros and might also need to be fixed upstream in jemallocator.

I think for now we can simply update the code comment to point to #9616 and retain the Windows check here. WDYT?

Here is the full log for the failure on NixOS
```
   Compiling tokio-macros v2.4.0
error: failed to run custom build command for `tikv-jemalloc-sys v0.6.0+5.3.0-1-ge13ca993e8ccb9ba9847cc330696e02839f328f7`

Caused by:
  process didn't exit successfully: `/home/mukilan/dev/servo.git/servo path with space/target/release/build/tikv-jemalloc-sys-4858c9903f489858/build-script-build` (exit status: 101)
  --- stdout
  TARGET=x86_64-unknown-linux-gnu
  HOST=x86_64-unknown-linux-gnu
  NUM_JOBS=24
  OUT_DIR="/home/mukilan/dev/servo.git/servo path with space/target/release/build/tikv-jemalloc-sys-7457db5be1c99ccd/out"
  BUILD_DIR="/home/mukilan/dev/servo.git/servo path with space/target/release/build/tikv-jemalloc-sys-7457db5be1c99ccd/out/build"
  SRC_DIR="/home/mukilan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tikv-jemalloc-sys-0.6.0+5.3.0-1-ge13ca993e8ccb9ba9847cc330696e02839f328f7"
  cargo:rustc-cfg=prefixed
  cargo:rerun-if-env-changed=X86_64_UNKNOWN_LINUX_GNU_JEMALLOC_OVERRIDE
  cargo:rerun-if-env-changed=JEMALLOC_OVERRIDE
  OPT_LEVEL = Some(3)
  TARGET = Some(x86_64-unknown-linux-gnu)
  OUT_DIR = Some(/home/mukilan/dev/servo.git/servo path with space/target/release/build/tikv-jemalloc-sys-7457db5be1c99ccd/out)
  HOST = Some(x86_64-unknown-linux-gnu)
  cargo:rerun-if-env-changed=CC_x86_64-unknown-linux-gnu
  CC_x86_64-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CC_x86_64_unknown_linux_gnu
  CC_x86_64_unknown_linux_gnu = None
  cargo:rerun-if-env-changed=HOST_CC
  HOST_CC = None
  cargo:rerun-if-env-changed=CC
  CC = Some(clang)
  RUSTC_WRAPPER = None
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some(false)
  cargo:rerun-if-env-changed=CFLAGS_x86_64-unknown-linux-gnu
  CFLAGS_x86_64-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CFLAGS_x86_64_unknown_linux_gnu
  CFLAGS_x86_64_unknown_linux_gnu = None
  cargo:rerun-if-env-changed=HOST_CFLAGS
  HOST_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  CC="clang"
  CFLAGS="-O3 -ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-unknown-linux-gnu -Wall"
  JEMALLOC_REPO_DIR="jemalloc"
  cargo:rerun-if-env-changed=X86_64_UNKNOWN_LINUX_GNU_JEMALLOC_SYS_WITH_MALLOC_CONF
  cargo:rerun-if-env-changed=JEMALLOC_SYS_WITH_MALLOC_CONF
  cargo:rerun-if-env-changed=X86_64_UNKNOWN_LINUX_GNU_JEMALLOC_SYS_WITH_LG_PAGE
  cargo:rerun-if-env-changed=JEMALLOC_SYS_WITH_LG_PAGE
  cargo:rerun-if-env-changed=X86_64_UNKNOWN_LINUX_GNU_JEMALLOC_SYS_WITH_LG_HUGEPAGE
  cargo:rerun-if-env-changed=JEMALLOC_SYS_WITH_LG_HUGEPAGE
  cargo:rerun-if-env-changed=X86_64_UNKNOWN_LINUX_GNU_JEMALLOC_SYS_WITH_LG_QUANTUM
  cargo:rerun-if-env-changed=JEMALLOC_SYS_WITH_LG_QUANTUM
  cargo:rerun-if-env-changed=X86_64_UNKNOWN_LINUX_GNU_JEMALLOC_SYS_WITH_LG_VADDR
  cargo:rerun-if-env-changed=JEMALLOC_SYS_WITH_LG_VADDR
  --with-jemalloc-prefix=_rjem_
  CARGO_FEATURE_STATS not set
  running: cd "/home/mukilan/dev/servo.git/servo path with space/target/release/build/tikv-jemalloc-sys-7457db5be1c99ccd/out/build" && CC="clang" CFLAGS="-O3 -ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-unknown-linux-gnu -Wall" CPPFLAGS="-O3 -ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-unknown-linux-gnu -Wall" LDFLAGS="-O3 -ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-unknown-linux-gnu -Wall" "sh" "/home/mukilan/dev/servo.git/servo path with space/target/release/build/tikv-jemalloc-sys-7457db5be1c99ccd/out/build/configure" "--with-version=5.3.0-1-ge13ca993e8ccb9ba9847cc330696e02839f328f7" "--disable-cxx" "--enable-doc=no" "--enable-shared=no" "--with-jemalloc-prefix=_rjem_" "--with-private-namespace=_rjem_" "--disable-stats" "--host=x86_64-unknown-linux-gnu" "--build=x86_64-unknown-linux-gnu" "--prefix=/home/mukilan/dev/servo.git/servo path with space/target/release/build/tikv-jemalloc-sys-7457db5be1c99ccd/out"
  running: "tail" "-n" "100" "/home/mukilan/dev/servo.git/servo path with space/target/release/build/tikv-jemalloc-sys-7457db5be1c99ccd/out/build/config.log"
  XSLROOT=''
  XSLTPROC=''
  a=''
  abi=''
  abs_objroot='/home/mukilan/dev/servo.git/servo path with space/target/release/build/tikv-jemalloc-sys-7457db5be1c99ccd/out/build/'
  abs_srcroot='/home/mukilan/dev/servo.git/servo path with space/target/release/build/tikv-jemalloc-sys-7457db5be1c99ccd/out/build/'
  ac_ct_CC=''
  ac_ct_CXX=''
  bindir='${exec_prefix}/bin'
  build='x86_64-unknown-linux-gnu'
  build_alias='x86_64-unknown-linux-gnu'
  build_cpu=''
  build_os=''
  build_vendor=''
  cfghdrs_in=''
  cfghdrs_out=''
  cfgoutputs_in=''
  cfgoutputs_out=''
  datadir='${datarootdir}'
  datarootdir='${prefix}/share'
  docdir='${datarootdir}/doc/${PACKAGE}'
  dvidir='${docdir}'
  enable_autogen=''
  enable_cache_oblivious=''
  enable_cxx='no'
  enable_debug=''
  enable_doc='no'
  enable_experimental_smallocx=''
  enable_fill=''
  enable_initial_exec_tls=''
  enable_lazy_lock=''
  enable_log=''
  enable_opt_safety_checks=''
  enable_opt_size_checks=''
  enable_prof=''
  enable_readlinkat=''
  enable_shared='no'
  enable_static=''
  enable_stats='no'
  enable_tls=''
  enable_uaf_detection=''
  enable_utrace=''
  enable_xmalloc=''
  enable_zone_allocator=''
  exe=''
  exec_prefix='NONE'
  host='x86_64-unknown-linux-gnu'
  host_alias='x86_64-unknown-linux-gnu'
  host_cpu=''
  host_os=''
  host_vendor=''
  htmldir='${docdir}'
  importlib=''
  includedir='${prefix}/include'
  infodir='${datarootdir}/info'
  install_suffix=''
  je_=''
  jemalloc_version=''
  jemalloc_version_bugfix=''
  jemalloc_version_gid=''
  jemalloc_version_major=''
  jemalloc_version_minor=''
  jemalloc_version_nrev=''
  libdir='${exec_prefix}/lib'
  libdl=''
  libexecdir='${exec_prefix}/libexec'
  libprefix=''
  link_whole_archive=''
  localedir='${datarootdir}/locale'
  localstatedir='${prefix}/var'
  mandir='${datarootdir}/man'
  o=''
  objroot=''
  oldincludedir='/usr/include'
  pdfdir='${docdir}'
  prefix='/home/mukilan/dev/servo.git/servo path with space/target/release/build/tikv-jemalloc-sys-7457db5be1c99ccd/out'
  private_namespace=''
  program_transform_name='s,x,x,'
  psdir='${docdir}'
  rev='2'
  sbindir='${exec_prefix}/sbin'
  sharedstatedir='${prefix}/com'
  so=''
  srcroot=''
  sysconfdir='${prefix}/etc'
  target_alias=''

  ## ----------- ##
  ## confdefs.h. ##
  ## ----------- ##

  /* confdefs.h */
  #define PACKAGE_NAME ""
  #define PACKAGE_TARNAME ""
  #define PACKAGE_VERSION ""
  #define PACKAGE_STRING ""
  #define PACKAGE_BUGREPORT ""
  #define PACKAGE_URL ""

  configure: exit 1

  --- stderr
  configure: error: Prefix should not contain spaces
  thread 'main' panicked at /home/mukilan/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tikv-jemalloc-sys-0.6.0+5.3.0-1-ge13ca993e8ccb9ba9847cc330696e02839f328f7/build.rs:388:9:
  command did not execute successfully: cd "/home/mukilan/dev/servo.git/servo path with space/target/release/build/tikv-jemalloc-sys-7457db5be1c99ccd/out/build" && CC="clang" CFLAGS="-O3 -ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-unknown-linux-gnu -Wall" CPPFLAGS="-O3 -ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-unknown-linux-gnu -Wall" LDFLAGS="-O3 -ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-unknown-linux-gnu -Wall" "sh" "/home/mukilan/dev/servo.git/servo path with space/target/release/build/tikv-jemalloc-sys-7457db5be1c99ccd/out/build/configure" "--with-version=5.3.0-1-ge13ca993e8ccb9ba9847cc330696e02839f328f7" "--disable-cxx" "--enable-doc=no" "--enable-shared=no" "--with-jemalloc-prefix=_rjem_" "--with-private-namespace=_rjem_" "--disable-stats" "--host=x86_64-unknown-linux-gnu" "--build=x86_64-unknown-linux-gnu" "--prefix=/home/mukilan/dev/servo.git/servo path with space/target/release/build/tikv-jemalloc-sys-7457db5be1c99ccd/out"
```

Reply:
done in ` dd4d03e`